test/e2e/test-data/otelcollector.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: otel-collector-conf
+  labels:
+    app: opentelemetry
+    component: otel-collector-conf
+data:
+  otel-collector-config: |
+    receivers:
+      otlp:
+        protocols:
+          http:
+
+    exporters:
+      logging:
+        loglevel: debug
+      prometheus:
+        endpoint: "0.0.0.0:8889"
+        send_timestamps: true
+        metric_expiration: 180m
+
+    service:
+      telemetry:
+        logs:
+          encoding: json
+      pipelines:
+        metrics:
+          receivers:
+            - otlp
+          exporters:
+            - logging
+            - prometheus
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: otel-collector
+  labels:
+    app: opentelemetry
+    component: otel-collector
+spec:
+  ports:
+  - name: otlp-http # Default endpoint for OpenTelemetry HTTP receiver.
+    port: 4318
+    protocol: TCP
+  - name: prometheus
+    port: 8889
```

Comment:
FYI if you use
```yaml
port: 80
targetPort: 8889
```
then you can get away with using `http://otel-collector/metrics` as the endpoint. It will listen on port 80, and send that traffic to 8889 on the actual pod.